### DISCUSSION TITLE
Confused about the removal of the action function

### DISCUSSION THREAD

**Discussion by akowi-sknobloch** [ai/nanostores#281] [General]
Im a bit confused why the `action()` function was removed in the `0.10.0` Release.
The Release only states that it was removed but not why.
I suppose because the functionality can also be achieved through normal functions?

Rewriting the actions as normal functions is not a big problem.
But the removal also breaks every function that was listening for a specific action.
Also the `@nanostores/logger` does depend on the old action style.
I can't see a way to achieve the same behaviour with the "normal functions" approach.

**Comment by ai** [comment]
> I suppose because the functionality can also be achieved through normal functions?

Yes

> Also the @nanostores/logger does depend on the old action style.

Send PR there

> But the removal also breaks every function that was listening for a specific action.

Do you want to create a `nanostores/action` for that? I found that `action` is too complex and not very useful.

**Comment by euaaaio** [comment]
> Im a bit confused why the `action()` function was removed in the `0.10.0` Release. The Release only states that it was removed but not why. I suppose because the functionality can also be achieved through normal functions?
> 
> Rewriting the actions as normal functions is not a big problem. But the removal also breaks every function that was listening for a specific action. Also the `@nanostores/logger` does depend on the old action style. I can't see a way to achieve the same behaviour with the "normal functions" approach.

Actions API will be added to `@nanostores/logger` this week.
